File: services/HospitalService/src/main/java/com/example/hospitalservice/service/MedicalRecordService.java
package com.example.hospitalservice.service;

import com.example.hospitalservice.entities.MedicalRecord;

import java.util.List;

public interface MedicalRecordService {
    MedicalRecord createRecord(MedicalRecord record);
    MedicalRecord updateRecord(Long id, MedicalRecord record);
    void deleteRecord(Long id);
    MedicalRecord getRecordById(Long id);
    List<MedicalRecord> getRecordsByPatient(Long patientId);
}

File: frontend/src/app/front-doctor/services/medical-record.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { MedicalRecord } from '../models/medical-record.model';

@Injectable({ providedIn: 'root' })
export class MedicalRecordService {
  private apiUrl = 'http://localhost:8082/hospitalService/api/medicalRecords';

  constructor(private http: HttpClient) {}

  getRecordsByPatient(patientId: number): Observable<MedicalRecord[]> {
    return this.http.get<MedicalRecord[]>(`${this.apiUrl}/patient/${patientId}`);
  }

  getRecordById(id: number): Observable<MedicalRecord> {
    return this.http.get<MedicalRecord>(`${this.apiUrl}/${id}`);
  }

  createRecord(record: MedicalRecord): Observable<MedicalRecord> {
    return this.http.post<MedicalRecord>(`${this.apiUrl}/create`, record);
  }

  updateRecord(id: number, record: MedicalRecord): Observable<MedicalRecord> {
    return this.http.put<MedicalRecord>(`${this.apiUrl}/update/${id}`, record);
  }

  deleteRecord(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/delete/${id}`);
  }
}
